lib/cookie-support.ts

```diff
@@ -75,4 +89,24 @@ export class CookieInterceptor {
     }
     return response;
   }
+
+  /**
+   * This is called by Axios when a non-2xx response has been received.
+   * We'll simply invoke the "responseFulfilled" method since we want to
+   * do the same cookie handler as for a success response.
+   * @param error the Axios error object that describes the non-2xx response
+   * @returns the error object
+   */
+  public async responseRejected(error: any) {
+    logger.debug('CookieIntercepter: intercepting error response');
+
+    if (error && error.response) {
```

Comment:
I recommend to write

```ts
    if (isAxiosError(error)) {
```

instead, because this gives you type safety on the `error.response` object